src/comfortable-swipe.cpp

```diff
@@ -186,36 +188,17 @@ namespace service {
     // get the full path of the .conf file
     string conf_filename() {
         static string *filename = NULL;
-        if (filename == NULL) {
-            const char* xdg_config = getenv("XDG_CONFIG_HOME");
-            string config(
-                xdg_config == NULL
-                    ? string(getenv("HOME")) + "/.config"
-                    : xdg_config
-            );
-            filename = new string(config + "/comfortable-swipe.conf");
-        }
-        return *filename;
-    }
-    // get the full path of the .desktop file associated
-    // with the autostart feature
-    string autostart_filename() {
-        static string *filename = NULL;
-        if (filename == NULL) {
-            const char* xdg_config = getenv("XDG_CONFIG_HOME");
-            string config(
-                xdg_config == NULL
-                    ? string(getenv("HOME")) + "/.config"
-                    : xdg_config
-            );
-            filename = new string(config
-                + "/autostart/comfortable-swipe.desktop");
-        }
+        if (filename == NULL)
+            filename = new string("/usr/local/share/comfortable-swipe.conf");
         return *filename;
     }
 }
 
-namespace service {    
+namespace service {
+    void exec() {
+        sleep(1);
```

Comment:
>@schigum in issue #16
>
>Hey, thanks so much for providing this.
>It is exactly what I was looking for.
>
>I'm running this on Ubuntu 17.10.
>One oddity I noticed is that I have a delay of ~1s before the swipe gets executed and the output in >console appears (e.g. SWIPE down 4).
>So I (quickly) swipe, take the fingers of the pad and after about 1s it gets executed.
>The main question is, is that normal? And can anything be done about that?
>
>Thanks and cheers!

Indeed, the cause of this is the `sleep(1)` which delays 1 second before executing. This was intended delay for the autostart, but I guess your point makes sense that there's no such need when one just wants to run it normally.

I'm looking for ways to activate it without the delay (using purely dependencies), but I still can't find an appropriate line of code for this to work. If it helps, the current service unit looks like:

```
[Unit]
Description=Comfortable 3 or 4 finger gestures
After=display-manager.service keyboard-setup.service

[Service]
Environment=DISPLAY=$DISPLAY XAUTHORITY=$XAUTHORITY
ExecStart=/usr/local/bin/comfortable-swipe exec
Restart=always
RestartSec=2

[Install]
WantedBy=graphical.target
```

Reply:
I'm not sure if I fully understand.
I'm using the master branch (sorry for not clarifying earlier). So there's no sleep(1) in the cpp file.
Was that introduced to "workaround" the mentioned delay?
To me it sounds like the delay in the code concerns the start of the program. The delay I mentioned happens for the execution of a swipe while the program is running.

Reply:
Sorry for misunderstanding, @schigum , I thought it was related to this PR, but it seems I was mistaken. Will reopen issue #16